stage1/init/kvm/network.go

```diff
@@ -33,36 +40,125 @@ func GetNetworkDescriptions(n *networking.Networking) []netDescriber {
 
 // netDescriber is something that describes network configuration
 type netDescriber interface {
-	HostIP() net.IP
 	GuestIP() net.IP
 	Mask() net.IP
 	IfName() string
 	IPMasq() bool
+	Name() string
+	Gateway() net.IP
+	Routes() []types.Route
 }
 
-// GetKVMNetArgs returns additional arguments that need to be passed to kernel
-// and lkvm tool to configure networks properly.
+// GetKVMNetArgs returns additional arguments that need to be passed
+// to lkvm tool to configure networks properly.
 // Logic is based on Network configuration extracted from Networking struct
 // and essentially from activeNets that expose netDescriber behavior
-func GetKVMNetArgs(nds []netDescriber) ([]string, []string, error) {
+func GetKVMNetArgs(nds []netDescriber) ([]string, error) {
 
 	var lkvmArgs []string
-	var kernelParams []string
-
-	for i, nd := range nds {
-		// https://www.kernel.org/doc/Documentation/filesystems/nfs/nfsroot.txt
-		// ip=<client-ip>:<server-ip>:<gw-ip>:<netmask>:<hostname>:<device>:<autoconf>:<dns0-ip>:<dns1-ip>
-		var gw string
-		if nd.IPMasq() {
-			gw = nd.HostIP().String()
-		}
-		kernelParam := fmt.Sprintf("ip=%s::%s:%s::%s:::", nd.GuestIP(), gw, nd.Mask(), fmt.Sprintf(networking.IfNamePattern, i))
-		kernelParams = append(kernelParams, kernelParam)
 
+	for _, nd := range nds {
 		lkvmArgs = append(lkvmArgs, "--network")
-		lkvmArg := fmt.Sprintf("mode=tap,tapif=%s,host_ip=%s,guest_ip=%s", nd.IfName(), nd.HostIP(), nd.GuestIP())
+		lkvmArg := fmt.Sprintf("mode=tap,tapif=%s,host_ip=%s,guest_ip=%s", nd.IfName(), nd.Gateway(), nd.GuestIP())
```

Comment:
Is `nd.Gateway()` necessarily the host? I'm thinking of cases where a user might define a different gateway for example with the bridge plugin